Only case sensitive check for minimumLevel override

**Description**
Serilog currently only supports case sensitive matches when overriding the log level for a specific namespace, when for example using Serilog with Serilog.Settings.Configuration and setting env variables for a docker image the Namespaces can't be all upper case.

**Reproduction**
Setting up a test project with Serilog and Serilog.Settings.Configuration installed, configure Serilog for example like this:

``` C#
host.UseSerilog((context, configuration) =>
        {
            configuration
                .WriteTo.Console(new JsonFormatter())
                .ReadFrom.Configuration(context.Configuration);
        })
```

add a appsettings file with the following configuration:

``` json
"Serilog": {
    "Using":  [ "Serilog.Sinks.Console" ],
    "MinimumLevel": {
      "Default": "Information",
      "Override": {
        "Test.Nmspc": "Information"
      }
    },
    "WriteTo": [
      { "Name": "Console" }
    ],
    "Enrich": [
      "FromLogContext"
    ],
    "Properties": {
      "Application": "Org.Eclipse.TractusX.Portal.Backend.Administration.Service"
    }
  },
```

Then execute the application with the following env var: SERILOG__MINIMUMLEVEL__OVERRIDE__TEST.NMSPC=Debug


The minimum logLevel for Loggers within that namespace will be Information

If needed I can provide an example repo.

**Expected behavior**
The minimum log level for logger instances creates within the Test.Nmspc Namespace should be Debug.

**Relevant package, tooling and runtime versions**
- Serilog 3.0.1
- Serilog.Settings.Configuration 3.4.0
- aspnet core 6 & 7 (both tested)
- docker image with environment variables (tested locally with a secret file as well on a win 10 machine)

**Additional context**
I've already provided a PR with a change for the case insensitive check.

Hmmm. I'm going to have to work this one out. Having trouble with BenchmarkDotNet.

`dotnet test -c Release -f net8.0 --filter "FullyQualifiedName=Serilog.PerformanceTests.Harness.SourceContextMatch"`

Trying to run that, I get a bunch of errors about how the autogenerated benchmark assembly is tareting netcoreapp2.1 and that there's a security vulnerability in `Microsoft.NETCore.App` 2.1.0.

```text
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
No test matches the given testcase filter `FullyQualifiedName=Serilog.PerformanceTests.Harness.SourceContextMatch` in /Users/travis.illig/dev/tillig/serilog/test/Serilog.ApprovalTests/bin/Release/net8.0/Serilog.ApprovalTests.dll

No test matches the given testcase filter `FullyQualifiedName=Serilog.PerformanceTests.Harness.SourceContextMatch` in /Users/travis.illig/dev/tillig/serilog/test/Serilog.Tests/bin/Release/net8.0/Serilog.Tests.dll

// Validating benchmarks:
// ***** BenchmarkRunner: Start   *****
// ***** Found 6 benchmark(s) in total *****
// ***** Building 2 exe(s) in Parallel: Start   *****
// ***** Done, took 00:00:01 (1.3 sec)   *****
// ***** Failed to build in Parallel, switching to sequential build   *****
// ***** Done, took 00:00:01 (1.98 sec)   *****
// Found 6 benchmarks:
//   SourceContextMatchBenchmark.Filter_MatchingFromSource: .NET Core 2.1(Runtime=.NET Core 2.1)
//   SourceContextMatchBenchmark.Logger_ForContext: .NET Core 2.1(Runtime=.NET Core 2.1)
//   SourceContextMatchBenchmark.LevelOverrideMap_GetEffectiveLevel: .NET Core 2.1(Runtime=.NET Core 2.1)
//   SourceContextMatchBenchmark.Filter_MatchingFromSource: .NET Core 3.1(Runtime=.NET Core 3.1)
//   SourceContextMatchBenchmark.Logger_ForContext: .NET Core 3.1(Runtime=.NET Core 3.1)
//   SourceContextMatchBenchmark.LevelOverrideMap_GetEffectiveLevel: .NET Core 3.1(Runtime=.NET Core 3.1)

// Build Error: Standard output:

 Standard error:
   Determining projects to restore...
/Users/travis.illig/dev/tillig/serilog/test/Serilog.PerformanceTests/bin/Release/net8.0/b43c468c-6c15-4641-87cb-afcb8321e852/BenchmarkDotNet.Autogenerated.csproj : warning NU1903: Package 'Microsoft.NETCore.App' 2.1.0 has a known high severity vulnerability, https://github.com/advisories/GHSA-2xjx-v99w-gqf3
```

That causes the benchmark to fail to compile and run. I may need to change this to use [the BenchmarkSwitcher](https://benchmarkdotnet.org/articles/guides/how-to-run.html) in my branch.

--- a/src/Serilog/Core/LevelOverrideMap.cs
+++ b/src/Serilog/Core/LevelOverrideMap.cs
@@ -68,7 +68,7 @@ class LevelOverrideMap
     {
         foreach (var levelOverride in _overrides)
         {
-            if (context.StartsWith(levelOverride.Context) &&
+            if (context.StartsWith(levelOverride.Context, StringComparison.OrdinalIgnoreCase) &&
                 (context.Length == levelOverride.Context.Length || context[levelOverride.Context.Length] == '.'))
             {
                 minimumLevel = LevelAlias.Minimum;

--- a/src/Serilog/Core/LevelOverrideMap.cs
+++ b/src/Serilog/Core/LevelOverrideMap.cs
@@ -68,7 +68,11 @@ class LevelOverrideMap
     {
         foreach (var levelOverride in _overrides)
         {
-            if ((context.Length == levelOverride.Context.Length || context[levelOverride.Context.Length] == '.') &&
+            if (
+                (
+                    context.Length == levelOverride.Context.Length ||
+                    (context.Length > levelOverride.Context.Length && context[levelOverride.Context.Length] == '.')
+                ) &&
                 context.StartsWith(levelOverride.Context, StringComparison.OrdinalIgnoreCase))
             {
                 minimumLevel = LevelAlias.Minimum;

--- a/src/Serilog/Core/LevelOverrideMap.cs
+++ b/src/Serilog/Core/LevelOverrideMap.cs
@@ -68,8 +68,8 @@ class LevelOverrideMap
     {
         foreach (var levelOverride in _overrides)
         {
-            if (context.StartsWith(levelOverride.Context, StringComparison.OrdinalIgnoreCase) &&
-                (context.Length == levelOverride.Context.Length || context[levelOverride.Context.Length] == '.'))
+            if ((context.Length == levelOverride.Context.Length || context[levelOverride.Context.Length] == '.') &&
+                context.StartsWith(levelOverride.Context, StringComparison.OrdinalIgnoreCase))
             {
                 minimumLevel = LevelAlias.Minimum;
                 levelSwitch = levelOverride.LevelSwitch;

--- a/test/Serilog.Tests/Core/LevelOverrideMapTests.cs
+++ b/test/Serilog.Tests/Core/LevelOverrideMapTests.cs
@@ -12,6 +12,9 @@ public class LevelOverrideMapTests
     [InlineData("MyApp.Api.Controllers.AboutController", true, Information)]
     [InlineData("MyApp.Api.Controllers.HomeController", true, Warning)]
     [InlineData("Api.Controllers.HomeController", false, LevelAlias.Minimum)]
+    [InlineData("serilog", false, LevelAlias.Minimum)]
+    [InlineData("myapp", true, Debug)]
+    [InlineData("myappsomething", false, LevelAlias.Minimum)]
     public void OverrideScenarios(string context, bool overrideExpected, LogEventLevel expected)
     {
         var overrides = new Dictionary<string, LoggingLevelSwitch>

